segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLog.java

```diff
@@ -115,7 +116,8 @@
         this.closed = new AtomicBoolean();
         this.logNodePath = HierarchyUtils.getPath(logId, this.config.getZkHierarchyDepth());
         this.traceObjectId = String.format("Log[%d]", logId);
-        this.writes = new WriteQueue(this.config.getMaxConcurrentWrites());
+        this.writes = new WriteQueue();
```

Comment:
I wanted to decouple the write queue from the concurrency manager - it makes it much easier to test. It also keeps the classes focused on what they are meant to do; one is a queue from which we pick items, and one keeps track of stats and determines parallelism.